pkg/internalregistry/controllers/service_account_controller.go

```diff
@@ -165,9 +198,10 @@ func (c *serviceAccountController) sync(ctx context.Context, key string) error {
 	var secretOwnerRefNeedsUpdate, secretSARefNeedsUpdate bool
 	if secret != nil {
 		secretSARefNeedsUpdate = secret.Annotations[InternalRegistryAuthTokenServiceAccountAnnotation] != serviceAccount.Name
+		secretOwnerRefNeedsUpdate = true
 		for _, ref := range secret.OwnerReferences {
 			if ref.Name == serviceAccount.Name && ref.UID == serviceAccount.UID && ref.Kind == "ServiceAccount" && ref.APIVersion == "v1" {
-				secretOwnerRefNeedsUpdate = true
+				secretOwnerRefNeedsUpdate = false
```

Comment:
Why are we flipping this from `true` to `false`? This inverts the logic of the for loop, while keeping the break condition unchanged.

Reply:
Logic was previously inverted. What we are trying to say here is: If the secret contains the service account as an owner already, we do not need to update the secret.